### PR TITLE
executor: implement NextChunk interface for UnionScanExec

### PR DESCRIPTION
**Pull request by zz-jason**
here I haven't implement a new mechanism to support Chunk for UnionScanExec, the reason is `dirtyTable` is write by Insert, Update, Replace and read by UnionScanExec

We have to support Chunk for dirtyTable and use these Chunk-oriented interfaces for Insert, Update, Replace and UnionScan in a single pull request.

to #5261

### CODE REVIEW

File: executor/union_scan.go [hunk 1]
```diff
@@ -108,6 +108,31 @@ type UnionScanExec struct {
...
 func (us *UnionScanExec) Next(goCtx goctx.Context) (Row, error) {
+	row, err := us.getOneRow(goCtx)
+	return row, errors.Trace(err)
+}
+
+// NextChunk implements the Executor NextChunk interface.
+func (us *UnionScanExec) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	for i, batchSize := 0, us.ctx.GetSessionVars().MaxChunkSize; i < batchSize; i++ {
+		row, err := us.getOneRow(goCtx)
```
Comment (winoros) [inline executor/union_scan.go]: `getOneRow` will call `getSnapshotRow`, `getSnapshotRow` will call `Next` method of child.

**Comment by shenli** [comment]
LGTM

File: executor/join_result_generators.go [hunk 2]
```diff
@@ -105,38 +104,7 @@ type baseJoinResultGenerator struct {
 func (outputer *baseJoinResultGenerator) initDefaultChunkInner(innerTypes []*types.FieldType) {
 	shadowChunk := chunk.NewChunk(innerTypes)
```
Comment (coocood) [inline executor/join_result_generators.go]: We can replace the `shadowChunk` with `chunk.MutRow`.

File: executor/union_scan.go [hunk 3]
```diff
@@ -108,6 +108,31 @@ type UnionScanExec struct {
...
 // Next implements Execution Next interface.
 func (us *UnionScanExec) Next(goCtx goctx.Context) (Row, error) {
+	row, err := us.getOneRow(goCtx)
+	return row, errors.Trace(err)
+}
+
+// NextChunk implements the Executor NextChunk interface.
+func (us *UnionScanExec) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+	chk.Reset()
+	mutableRow := chunk.MutRowFromTypes(us.Schema().GetTypes())
```
Comment (coocood) [inline executor/union_scan.go]: Make `mutableRow` a member of `UnionScanExec` is more efficient.

Comment (zz-jason) [reply executor/union_scan.go]: The usage of `mutableRow` is temporary, I think it's meaningless to consider the performance here. Once we have supported Chunk in dirty table, we can refactor it in the most efficient way.

**Pull request merged**